examples/Go/ChatApp/main.go

```diff
@@ -0,0 +1,224 @@
+	"github.com/openai/openai-go/azure"
+)
+
+type ChatApp struct {
+	configProvider    *azureappconfiguration.AzureAppConfiguration
+	openAIClient openai.Client
+	aiConfig     AIConfig
+}
+
+type AIConfig struct {
+	ChatCompletion ChatCompletion
+	AzureOpenAI    AzureOpenAI
+}
+
+type ChatCompletion struct {
+	Model       string    `json:"model"`
+	Messages    []Message `json:"messages"`
+	MaxTokens   int64     `json:"max_tokens"`
+	Temperature float64   `json:"temperature"`
+	TopP        float64   `json:"top_p"`
+}
+
+type AzureOpenAI struct {
+	Endpoint   string
+	APIVersion string
+	APIKey    string
+}
+
+type Message struct {
+	Role    string `json:"role"`
+	Content string `json:"content"`
+}
+
+func loadAzureAppConfiguration(ctx context.Context) (*azureappconfiguration.AzureAppConfiguration, error) {
+	endpoint := os.Getenv("AZURE_APPCONFIGURATION_ENDPOINT")
+	if endpoint == "" {
+		return nil, fmt.Errorf("AZURE_APPCONFIGURATION_ENDPOINT environment variable is not set")
+	}
+
+	credential, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Azure credential: %w", err)
+	}
+
+	authOptions := azureappconfiguration.AuthenticationOptions{
+		Endpoint:  endpoint,
+		Credential: credential,
+	}
+
+	options := &azureappconfiguration.Options{
+		Selectors: []azureappconfiguration.Selector{
+			// Load all keys that start with "ChatApp:" and have no label
+			{
+				KeyFilter:   "ChatApp:*",
+			},
+		},
+		TrimKeyPrefixes: []string{"ChatApp:"},
+		RefreshOptions: azureappconfiguration.KeyValueRefreshOptions{
+			Enabled:  true,
+			Interval: 10 * time.Second,
+		},
+		KeyVaultOptions: azureappconfiguration.KeyVaultOptions{
+			Credential: credential,
+		},
+	}
+
+	appConfig, err := azureappconfiguration.Load(ctx, authOptions, options)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load configuration: %w", err)
+	}
+
+	return appConfig, nil
+}
+
+// Create an Azure OpenAI client using API key if available, otherwise use the DefaultAzureCredential
+func (app *ChatApp) createAzureOpenAIClient() error {
+	if app.aiConfig.AzureOpenAI.APIKey != "" {
+		// Use API key for authentication
+		client := openai.NewClient(
+			azure.WithAPIKey(app.aiConfig.AzureOpenAI.APIKey),
+			azure.WithEndpoint(app.aiConfig.AzureOpenAI.Endpoint, app.aiConfig.AzureOpenAI.APIVersion),
+		)
+		app.openAIClient = client
+		return nil
+	}
+
+	// Use DefaultAzureCredential for authentication
+	tokenCredential, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return fmt.Errorf("failed to create Azure credential: %w", err)
+	}
+
+	client := openai.NewClient(
+		azure.WithEndpoint(app.aiConfig.AzureOpenAI.Endpoint, app.aiConfig.AzureOpenAI.APIVersion),
+		azure.WithTokenCredential(tokenCredential),
+	)
+
+	app.openAIClient = client
+	return nil
+}
+
+func (app *ChatApp) callAzureOpenAI(userMessage string) (string, error) {
+	messages := []openai.ChatCompletionMessageParamUnion{}
```

Comment:
The chat history should be stored and sent back to AI for the next round of the conversation. See .NET implementation:
https://github.com/Azure/AppConfiguration/blob/main/examples/DotNetCore/ChatApp/ChatApp/Program.cs#L57